Engine/SubscriptionDataReader.cs

```diff
@@ -554,23 +558,29 @@ private SubscriptionStreamReader GetReader(string source)
                     // check if this is not a local uri then download it to the local cache
                     if (uri.IsAbsoluteUri && !uri.IsLoopback)
                     {
-                        try
+                        //Skip download if the file is already in cache
+                        if (!File.Exists(location))
```

Comment:
Its safe for QC financial data which is static and known but not for user's custom CSVs. They can update them on Dropbox then we need to re-download them to use the updated data. Often the files are changed daily, or if they have an error they can fix it and run the backtest again to make it work. Quandl's data updates daily as well.